src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Handlers/RetryHandler.cs

```diff
@@ -74,17 +74,22 @@ public class RetryHandler : DelegatingHandler
                 }
                 catch (Exception ex)
                 {
-                    if (ex.IsFatal() || attempt == RetryHandler.MaxAttempts)
+                    if (ex.IsFatal() || attempt == RetryHandler.MaxAttempts - 1)
                     {
                         throw;
                     }
                 }
 
+                if (response != null)
+                {
+                    response.Dispose();
+                }
+
                 await Task.Delay(delay: RetryHandler.GetDelay(attempt), cancellationToken: cancellationToken)
                     .ConfigureAwait(continueOnCapturedContext: false);
             }
 
-            return response;
+            return null;
```

Comment:
The return is needed for this to successfully compile. I'd keep this as response (which is set to null either way), instead of returning null explicitly